worker_tasks/categorize.py

```diff
@@ -46,109 +80,374 @@
+
+                if link_title in sublinks:
+                    for sublink in sublinks[link_title]:
+                        if sublink not in self._memoized_article_links:
+                            self._memoized_article_links[sublink] = links[link_title]
+
+                # Store related articles
+                if link_title in disambiguation_pages:
+                    related_links = filter(lambda x: link_title.replace(" (disambiguation)", "").lower() in x.lower(), links[link_title])
+
+                    if link_title not in self._memoized_related_articles:
+                            self._memoized_related_articles[link_title] = related_links
+                            self._memoized_related_articles[link_title.replace(" (disambiguation)", "")] = related_links
+
+
+                    if link_title in sublinks:
+                        for sublink in sublinks[link_title]:
+                            if sublink not in self._memoized_related_articles:
+                                self._memoized_related_articles[sublink] = related_links
+                                self._memoized_related_articles[sublink.replace(" (disambiguation)", "")] = related_links
+                else:
+                    if link_title not in self._memoized_related_articles:
+                            self._memoized_related_articles[link_title] = [link_title]
+
+                    if link_title in sublinks:
+                        for sublink in sublinks[link_title]:
+                            if sublink not in self._memoized_related_articles:
+                                self._memoized_related_articles[sublink] = [link_title]
+
+
+    def _compile_sublinks(self, data, link_path=None):
+        """Flatten normalized and redirect links for quick lookup"""
+
+        link_redirects = defaultdict(set)
+        if "redirects" in data:
+            for redirect in data['redirects']:
+                link_redirects[redirect['to']].add(redirect['from'])
+
+        if not link_path:
+            link_path = link_redirects    
+
+        if "normalized" in data: 
+            for normalized_link in data['normalized']:
+                link_path[normalized_link['to']].add(normalized_link['from'])
+                
+                for to_link, from_links in link_path.iteritems():
+                    if normalized_link['to'] in from_links:
+                        link_path[to_link].update(normalized_link['to'], normalized_link['from'])
+
+        return link_path
+
+
+    def _decode_list(self, data):
+        rv = []
+        for item in data:
+            if isinstance(item, unicode):
+                item = item.encode('utf-8')
+            elif isinstance(item, list):
+                item = self._decode_list(item)
+            elif isinstance(item, dict):
+                item = self._decode_dict(item)
+            rv.append(item)
+        return rv
+
+
+    def _decode_dict(self, data):
```

Comment:
Since these two functions are pretty trivial, I tried to cover them all in as few tests as possible. The list contains a dict, which is how I got the coverage to pass with 100%. Figured it wasn't worth writing too many tests over.